pkg/services/tree/sync.go

```diff
@@ -303,12 +335,20 @@ func (s *Service) syncLoop(ctx context.Context) {
 			}
 			wg.Wait()
 
-			// remove stored redundant trees
+			s.cnrMapMtx.Lock()
+			var removed []cid.ID
 			for cnr := range s.cnrMap {
 				if _, ok := newMap[cnr]; ok {
 					continue
 				}
+				removed = append(removed, cnr)
+			}
+			for i := range removed {
+				delete(s.cnrMap, removed[i])
```

Comment:
we have one more `delete` without mutex above

Reply:
Fixed